clarity/src/vm/callables.rs

```diff
@@ -175,7 +179,13 @@ impl DefinedFunction {
             // Clarity 1 behavior
             if *env.contract_context.get_clarity_version() < ClarityVersion::Clarity2 {
                 match (type_sig, value) {
+                    // Epoch < 2.1 uses TraitReferenceType
```

Comment:
Should this `if`-branch also check the epoch?  There are three combinations allowed -- stacks 2.1 + clarity 2, stacks 2.1 + clarity 1, and stacks 2.05 (or lower) and clarity 1.

I think one of the invariants you're trying to preserve here is that only the 2.05-series type checker can ever instantiate a `TypeSignature::TraitReferenceType`?  Should callers simply expect this?

Reply:
Right, when in epoch < 2.1, we should not see `CallableType` and in epoch >= 2.1, we should not see `TraitReferenceType`. Yes, I could add a check to make sure the epoch matches the correct type here.